samcli/commands/check/resources/Pricing.py

```diff
@@ -0,0 +1,155 @@
+"""
+Pricing questions are asked here. Pricing is only done for
+Lambda Functions as of now. Data is stored in graph in
+Lambda function pricing object.
+"""
+
+from typing import List, Tuple
+import click
+
+from samcli.commands.check.resources.Graph import Graph
+from samcli.commands.check.resources.LambdaFunctionPricing import LambdaFunctionPricing
+from samcli.commands._utils.resources import AWS_LAMBDA_FUNCTION
+
+
+class Pricing:
+    _graph: Graph
+    _max_num_requests: int
+    _min_memory_amount: int
+    _max_memory_amount: int
+    _max_duration: int
+
+    def __init__(self, graph: Graph) -> None:
+        """
+        Args:
+            graph (Graph): The graph object. This is where all of the data is stored
+        """
+        self._graph: Graph = graph
+        self._max_num_requests: int = 1000000000000000000000
+        self._min_memory_amount: int = 128
+        self._max_memory_amount: int = 10000
+        self._max_duration: int = 900000
+
+    def ask_pricing_questions(self) -> None:
+        """
+        Pricing quetions for various resources get asked here
+        Pricing is only done for Lambda functions now
+        """
+        asked_lambda_questions = False
+        click.echo("Pricing Questions")
+        for resource in self._graph.resources_to_analyze:
+            # Only ask lambda quetions once for all lambda functions
+            if resource.resource_type == AWS_LAMBDA_FUNCTION and not asked_lambda_questions:
+                asked_lambda_questions = True
+                self._ask_lambda_function_questions()
+
+    def _ask_lambda_function_questions(self) -> None:
```

Comment:
I was thinking this class is specifically used for Lambda type since it has values of `max_num_requests`, `min_memory_amount`, but now I see `_ask_lambda_function_questions`. 

If this class is generic then it shouldn't contain properties specific to a resource type.
If this class is specific to a resource type, then it doesn't need `_ask_lambda_questions`.

In any case, let's target for something which is easily expandable. We should contain resource specific parameters in the resource specific classes. And when we need information for certain resource type, we can instantiate that class instance and use it to ask the questions.

Reply:
I see what you're saying. This class will eventually be for all resources, so I will extract the lambda pricing logic to its own class.